MagneticField/VolumeBasedEngine/src/MagGeometry.cc

```diff
@@ -199,15 +217,8 @@ MagVolume const* MagGeometry::findVolume(const GlobalPoint& gp, double tolerance
 }
 
 bool MagGeometry::inBarrel(const GlobalPoint& gp) const {
-  float Z = fabs(gp.z());
-  float R = gp.perp();
-
-  // FIXME: Get these dimensions from the builder.
-  if (geometryVersion >= 120812) {
-    return (Z < 350. || (R > 172.4 && Z < 633.29) || (R > 308.735 && Z < 662.01));
-  } else if (geometryVersion >= 90812) {  // FIXME no longer supported
-    return (Z < 350. || (R > 172.4 && Z < 633.89) || (R > 308.755 && Z < 662.01));
-  } else {  // versions 71212, 90322
-    return (Z < 350. || (R > 172.4 && Z < 633.29) || (R > 308.755 && Z < 661.01));
-  }
+  float aZ = fabs(gp.z());
+  float aRsq = gp.perp2();
+
+  return ((aZ < barrelZ0) || (aZ < barrelZ1 && aRsq > barrelRsq1) || (aZ < barrelZ2 && aRsq > barrelRsq2));
```

Comment:
the warnings mentioned in https://github.com/cms-sw/cmssw/pull/29556#issuecomment-620715989 are at 
r,z: (308.73498, 658.755) in the 2016 wf and (172.40009, 625.049) in the 2018 wf.
IIUC, we are hitting rounding issues.

Reply:
Indeed, the boundary values were doubles before, and I set them to float here. 
If I change boundaries back to doubless the problem goes away. But before claiming this is a fix I think I then need to do some more checks also for the change of making computations in perp2 instead than in perp. 
This is apparently not caught by regression which, I suppose, does not hits these points (standard regression is done comparing the output of 1M random points with a reference file)

Reply:
> Reco comparison results: 6 differences found in the comparisons

Slava, apologies for the dumb question, how do I find the other 4?

Reply:
this was a lazy copy-paste of the summary of diffs.
The differences are in two workflows as described in https://github.com/cms-sw/cmssw/pull/29556#issuecomment-620715989
there is nothing else, at least in reco steps.

Reply:
OK, these two are fixed by simply changing the constants back to doubles, but I clearly need to extend the regression until it catches these (and possible other) numerical side effects before proceeding. WIll do that tomorrow.

Reply:
OK I committed a fix, this is very simple but comes with some updates to test and a re-organization of MessageLogger categories that was necessary to be able to check whether these optimizations could also induce numerical problems elsewhere.
PS: I am porting this fix and all updates to #29561 as well.